src/convertToObject.js

```diff
@@ -6,7 +6,28 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.trim()) {
+    return {};
+  }
+
+  const result = {};
+
+  const rules = sourceString
+    .split(';')
+    .map((rule) => rule.trim())
+    .filter(Boolean);
```

Comment:
The `filter(Boolean)` method is used to remove any empty strings from the array. This is a good practice to ensure that only valid rules are processed.